Formula/odo-dev.rb

```diff
@@ -7,6 +7,11 @@ class OdoDev < Formula
   license "Apache-2.0"
   head "https://github.com/redhat-developer/odo.git", branch: "main"
 
+  livecheck do
```

Comment:
```suggestion
  # Upstream creates releases that use a stable tag (e.g., `v1.2.3`) but are
  # labeled as "pre-release" on GitHub before the version is released, so it's
  # necessary to use the `GithubLatest` strategy.
  livecheck do
```

Since this is using the `GithubLatest` strategy with a `stable` URL that's not a GitHub release asset (e.g., `.../releases/download/v1.2.3/example-1.2.3.tar.xz`), this needs a comment to explain the situation (so we don't mistakenly remove this `livecheck` block as unneeded).